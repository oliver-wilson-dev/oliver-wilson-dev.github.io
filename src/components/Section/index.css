@import "../../styles/transitions.css";

@keyframes fadeIn {
    0% { 
        opacity: 0; 
    }
    100% { 
        opacity: 1; 
    }
}

.section {
    transition: visibility var(--sectionVisibilityTransitionDuration), opacity var(--sectionOpacityTransitionDuration);
    transition-delay: var(--spinnerOpacityTransitionDuration);
    opacity: 0;
    visibility: hidden;
    &.show {
        opacity: 1;
        visibility: visible;
      }
}

.displayBox {
    background: var(--primaryBackground);
    -webkit-box-shadow: 8px 8px 0px var(--primaryBorderColour);
    -moz-box-shadow: 8px 8px 0px var(--primaryBorderColour);
    box-shadow: 8px 8px 0px var(--primaryBorderColour);
    border: 3px solid var(--primaryBorderColour);
    padding: 10px;
    max-height: 250px;
    overflow: hidden;
    position: relative;
}

.displayBoxExpanded {
    max-height: 1500px;
}

.collapseExpand {
    margin: 5px 5px 5px auto;
    position: absolute;
    right: 0px;
    top: 0px;
    font-size: 30px;
    width: 35px;
    border: none;
    background: transparent;
    font-family: inherit;
    font-weight: 900;
    color: inherit;
}

@media all and (min-width: 786px) {
    .landingBlurb {
        max-width: 800px;
    }
}