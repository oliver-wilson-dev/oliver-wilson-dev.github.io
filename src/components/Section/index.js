import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './index.css';
import sharedStyles from '../App/index.css';

const Section = ({ title, children }) => (
  <div className={cn(sharedStyles.flexColumn, sharedStyles.flexCenter)}>
    <h2 className={sharedStyles.sectionTitle}>{title}</h2>
    <div className={(cn(styles.landingBlurb, styles.displayBox))}>
      {children}
    </div>
  </div>
);

Section.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired
};

export default Section;
