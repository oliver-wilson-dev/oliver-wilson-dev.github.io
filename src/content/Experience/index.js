import React from 'react';
import classnames from 'classnames';
import styles from '../index.css';

export default {
  title: 'Experience',
  content:
  <div>
    <h3 className={styles.companyTitle}>ASOS</h3>
    <h4 className={styles.jobRole}>
      Associate Web Developer
    </h4>
    <h4>September 2017 - present</h4>
    <p>
      <span className={styles.paragraphEmoji} role="img" aria-label="university emoji">ðŸ’»</span>
        After being accepted for the first official graduate scheme for the web platform
        at asos I was catapulted into a number of projects that aided my personal
        and technical growth.
    </p>
    <p>
        During my time at asos I have been tasked with creating an internal react app that
        is used by the photography studios (the team that capture the product images displayed
        on the site) to coordinate their work and allow them to continue to deliver great assets
        while optimising their time and productivity.
    </p>
    <p>
        I have continued to maintain the Product Listing Page (PLP) which is an isomorphic app
        that responds to multiple viewports and is supported by a wide range of browsers.
        SEO is very important for the page as asos needs to stay ahead of its competitors and
        therefore rank higher in search results.
    </p>
    <p>
        I have been a key team member in the replatforming of the Saved Items application,
        employing a tech stack of react, redux and node,
        while using webpack to build the application as well as utilising babel and postcss
        to use the latest syntax available to developers.
    </p>
    <div className={styles.projectImages}>
      <div className={styles.projectImage}>
        <img alt="ASOS Product Listing Page" src="https://github.com/oliver-wilson-dev/oliver-wilson-dev.github.io/blob/master/public/img/plp.png?raw=true" />
      </div>
      <div className={classnames(styles.projectImage, styles.lastProjectImage)}>
        <img alt="ASOS Saved Lists Page" src="https://github.com/oliver-wilson-dev/oliver-wilson-dev.github.io/blob/master/public/img/saved-lists.png?raw=true" />
      </div>
    </div>
  </div>,
};
