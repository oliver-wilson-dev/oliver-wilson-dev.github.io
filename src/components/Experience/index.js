import React from 'react';
import PropTypes from 'prop-types';
import Section from '../Section';
import Workplace from '../Workplace';


const Experience = ({ title, workplaces, emoji }) => (
  <Section title={title}>
    {workplaces.map(workplace => (
      <Workplace
        key={workplace.companyName}
        {...workplace}
        emoji={emoji}
      />
    ))}
  </Section>
);

Experience.propTypes = {
  emoji: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  workplaces: PropTypes.array.isRequired
};

export default Experience;
