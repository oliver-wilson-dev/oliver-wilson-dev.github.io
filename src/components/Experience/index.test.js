import React from 'react';
import { shallow } from 'enzyme';
import Experience from '.';
import Section from '../Section';

const defaultProps = {
  companyName: 'test-company-name',
  content: ['test-paragraph', 'test-paragraph'],
  emoji: 'test-emoji',
  employmentPeriod: 'test-employment-period',
  jobTitle: 'test-job-title',
  title: 'test-title',
};

const render = overrideProps => shallow(<Experience {...defaultProps} {...overrideProps} />);

jest.mock('../Section', () => {
  const Section = () => null;

  return Section;
});

describe('<Experience/> component', () => {
  it('should exist', () => {
    expect(render().exists()).toBe(true);
  });

  it('should match snapshot', () => {
    expect(render()).toMatchSnapshot();
  });

  it('should render a Section component and pass it the "title" prop', () => {
    const testTitle = 'test-title';
    expect(render({ title: testTitle }).find(Section).prop('title')).toBe(testTitle);
  });
});
