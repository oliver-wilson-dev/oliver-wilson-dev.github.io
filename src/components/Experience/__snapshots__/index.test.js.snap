// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Experience/> component should match snapshot 1`] = `
<Section
  title="test-title"
>
  <h3
    className="companyTitle"
  >
    test-company-name
  </h3>
  <h4
    className="jobRole"
  >
    test-job-title
  </h4>
  <h4>
    test-employment-period
  </h4>
  <p
    key="test-paragraph"
  >
    <span
      aria-label="laptop emoji"
      className="paragraphEmoji"
      role="img"
    >
      test-emoji
    </span>
    test-paragraph
  </p>
  <p
    key="test-paragraph"
  >
    test-paragraph
  </p>
  <div
    className="projectImages"
  >
    <div
      className="projectImage"
    >
      <a
        href="https://www.asos.com/search/?q=jeans"
      >
        <img
          alt="ASOS Product Listing Page"
          src="https://github.com/oliver-wilson-dev/oliver-wilson-dev.github.io/blob/master/public/img/plp.png?raw=true"
        />
      </a>
    </div>
    <div
      className="projectImage lastProjectImage"
    >
      <a
        href="https://www.asos.com/saved-lists/"
      >
        <img
          alt="ASOS Saved Lists Page"
          src="https://github.com/oliver-wilson-dev/oliver-wilson-dev.github.io/blob/master/public/img/saved-lists.png?raw=true"
        />
      </a>
    </div>
  </div>
</Section>
`;
